src/qibolab/platform.py

```diff
@@ -121,14 +99,35 @@ def reload_settings(self):
             # dummy value to avoid transpiler failure for single qubit devices
             self.two_qubit_native_types = NativeType.CZ
 
-        if self.topology is None:
-            self.topology = nx.Graph()
-            self.topology.add_nodes_from(self.qubits.keys())
-            self.topology.add_edges_from([(pair.qubit1.name, pair.qubit2.name) for pair in self.pairs.values()])
+        self.topology: nx.Graph = nx.Graph()
+        self.topology.add_nodes_from(self.qubits.keys())
+        self.topology.add_edges_from([(pair.qubit1.name, pair.qubit2.name) for pair in self.pairs.values()])
```

Comment:
Maybe it is redundant but we could define a `Topology` class where you pass directly the qubits and qubit pairs.

Reply:
I think it is redundant since the `topology` is just a `nx.Graph` which is basically what the transpiler is using, where `qubits` and `pairs` are `platform` attributes. If the transpiler needs a more complicated structure we could consider creating another class but for now I don't think that's needed.

Another option would be to create a `QPU` class that has the qubits, pairs and topology, but I also think it's redundant at this stage. Instead of `platform.qubits` you'd have to do `platform.qpu.qubits` and I don't think we gain anything from this.

Reply:
If `QubitId`s were just numbers, we wouldn't even need a `qubits` attribute, since they could just be the nodes of the graph, maybe exposed through a property (actually, they could still be, since nodes can have metadata, so they could take the place of the dictionary, but it's not much simpler than the current setup...)